operator/pkg/helmreconciler/prune.go

```diff
@@ -78,6 +101,104 @@ func (h *HelmReconciler) Prune(manifests name.ManifestMap, all bool) error {
 	})
 }
 
+// PruneControlPlaneByRevisionWrapper is wrapper to remove specific control plane revision and update iop status.
+// It is used for the operator case.
+func (h *HelmReconciler) PruneControlPlaneByRevisionWrapper(ns, revision string) (*v1alpha1.InstallStatus, error) {
+	if ns == "" {
+		ns = istioDefaultNamespace
+	}
+	errStatus := &v1alpha1.InstallStatus{Status: v1alpha1.InstallStatus_ERROR}
+	pids, err := proxy.GetIDsFromProxyInfo("", "", revision, ns)
+	if err != nil {
+		return errStatus,
+			fmt.Errorf("failed to check proxy infos: %v", err)
```

Comment:
Maybe I am missing something but I don't think we have the message field correlated with status now? I am adding that in this api PR: https://github.com/istio/api/pull/1499. I added a TODO here for now

Reply:
from what i've seen the error message body shows up in the status field for the controller case. although that's not strictly necessary for uninstall command, we probably want to follow that convention in case this function ends up being used for controller case.